supervisor/jira_utils.py

```diff
@@ -57,16 +57,51 @@ def jira_url() -> str:
     return url.rstrip("/")
 
 
+@cache
+def jira_api_version() -> str:
+    if "JIRA_API_VERSION" in os.environ:
```

Comment:
This doesn't make much sense to me with this version of the patch - - it seems we always know what API version we want to call based on the API we are calling - and sometimes we pick *which* API we are calling based on whether we are using cloud or server.